### PR TITLE
Use Varargs in Model APIs

### PR DESCRIPTION
**Pull request by alzimmermsft**

### CODE REVIEW

File: sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/implementation/converters/AnalyzeRequestConverter.java [hunk 1]
```diff
@@ -35,15 +35,17 @@ public static AnalyzeTextOptions map(com.azure.search.documents.indexes.implemen
...
             analyzeTextOptions.setCharFilters(charFilters);
         }
 
 
         if (obj.getTokenFilters() != null) {
-            List<TokenFilterName> tokenFilters =
-                obj.getTokenFilters().stream().map(TokenFilterNameConverter::map).collect(Collectors.toList());
+            TokenFilterName[] tokenFilters = obj.getTokenFilters().stream()
+                .map(TokenFilterNameConverter::map)
+                .toArray(TokenFilterName[]::new);
```
Comment (srnagar) [inline sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/implementation/converters/AnalyzeRequestConverter.java:48]: These shouldn't have to change. Input-output models should have setter overloads for both varargs and List.

Comment (alzimmermsft) [reply sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/implementation/converters/AnalyzeRequestConverter.java]: Reverted

File: sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/indexes/models/AnalyzeTextOptions.java [hunk 2]
```diff
@@ -201,6 +204,20 @@ public LexicalTokenizerName getTokenizerName() {
      * @param tokenFilters the tokenFilters value to set.
      * @return the AnalyzeRequest object itself.
      */
+    public AnalyzeTextOptions setTokenFilters(TokenFilterName... tokenFilters) {
```
Comment (sima-zhu) [inline sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/indexes/models/AnalyzeTextOptions.java:207]: The options are input only

Comment (sima-zhu) [reply sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/indexes/models/AnalyzeTextOptions.java:207]: We can leave the varargs setter here and remove the list one.

Comment (alzimmermsft) [reply sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/indexes/models/AnalyzeTextOptions.java:206]: Removed

File: sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/indexes/models/CjkBigramTokenFilter.java [hunk 3]
```diff
@@ -58,6 +61,18 @@ public CjkBigramTokenFilter(@JsonProperty(value = "name") String name) {
      * @param ignoreScripts the ignoreScripts value to set.
      * @return the CjkBigramTokenFilter object itself.
      */
+    public CjkBigramTokenFilter setIgnoreScripts(CjkBigramTokenFilterScripts... ignoreScripts) {
+        this.ignoreScripts = Arrays.asList(ignoreScripts);
```
Comment (srnagar) [inline sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/indexes/models/CjkBigramTokenFilter.java:65]: This should have a null check. Update this in other places too.

```suggestion
        this.ignoreScripts = ignoreScripts == null ? null : Arrays.asList(ignoreScripts);
```

Comment (alzimmermsft) [reply sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/indexes/models/CjkBigramTokenFilter.java]: Added checks to all usage of varargs, additionally updated options models as well as they would run into the same issue.

**Pull request merged**